File: SEP3Java/src/main/java/com/via/sep3java/entity/BookRegistry.java
package com.via.sep3java.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

@Entity public class BookRegistry
{
  @Id private String isbn;

  @Column(nullable = false) private String title;

  @Column private String author;

  @Column private String description;

  @Column private String genre;

  @Column private String reviews;

  public String getGenre()
  {
    return genre;
  }

  public void setGenre(String genre)
  {
    this.genre = genre;
  }

  public String getReviews()
  {
    return reviews;
  }

  public void setReviews(String reviews)
  {
    this.reviews = reviews;
  }

  public BookRegistry()
  {
  }

  public String getDescription()
  {
    return description;
  }

  public void setDescription(String description)
  {
    this.description = description;
  }

  public String getTitle()
  {
    return title;
  }

  public void setTitle(String title)
  {
    this.title = title;
  }

  public String getAuthor()
  {
    return author;
  }

  public void setAuthor(String author)
  {
    this.author = author;
  }

  public String getIsbn()
  {
    return isbn;
  }

  public void setIsbn(String isbn)
  {
    this.isbn = isbn;
  }
}

File: SEP3CSharp/Domain/Models/BookRegistry.cs
using Newtonsoft.Json;

namespace Domain.Models;

public class BookRegistry
{
    public BookRegistry(string title, string author, string genre, string isbn, string description,
        string reviews)
    {
        Title = title;
        Author = author;
        Genre = genre;
        Isbn = isbn;
        Description = description;
        Reviews = reviews;
    }

    [JsonProperty("isbn")] public string Isbn { get; set; }

    [JsonProperty("title")] public string Title { get; set; }

    [JsonProperty("author")] public string Author { get; set; }

    [JsonProperty("genre")] public string Genre { get; set; }

    [JsonProperty("description")] public string Description { get; set; }

    [JsonProperty("reviews")] public string Reviews { get; set; }
}